push.sh

```diff
@@ -1,4 +1,36 @@
-#!/bin/sh
+#!/bin/bash
+#
+# Build image from the currently checked out version of
+# MusicBrainz Solr search server (mb-solr)
+# and push it to Docker Hub, tagged with version, and
+# aliases (major.minor and major).
+#
+# Usage:
+#   $ ./push.sh
 
-docker build -t metabrainz/solr .
-docker push metabrainz/solr
\ No newline at end of file
+set -e -u
+
+vcs_ref=`git describe --always --broken --dirty --tags`
+
+version=$(echo $vcs_ref | sed 's/^v-\?//')
+minor=$(echo $version | sed 's/^\([^.]\+\.[^.]\+\)\..\+$/\1/')
+major=$(echo $version | sed 's/\..*//')
+version_aliases=( $( (echo $minor; echo $major) | uniq | sed "/${version}/d" ) )
+
+variant=''
+tag=${version}${variant}
+tag_aliases=${version_aliases[@]/%/${variant}}
+
+cd "$(dirname "${BASH_SOURCE[0]}")/"
+
+docker build \
+  --build-arg MB_SOLR_VERSION=${version} \
+  --build-arg BUILD_DATE=`date -u +"%Y-%m-%dT%H:%M:%SZ"` \
+  --build-arg VCS_REF=${vcs_ref} \
+  --tag metabrainz/mb-solr:${tag} .
+docker push metabrainz/mb-solr:${tag}
```

Comment:
What about storing `metabrainz/mb-solr` in a variable (like `$image`) and use it.

Reply:
Thanks, done.